core/policy/service.go

```diff
@@ -217,7 +217,7 @@ func (s *Service) decryptAndDeserializeIAMConfig(c *domain.IAMConfig) error {
 	if err := iamClientConfig.Decrypt(); err != nil {
 		return fmt.Errorf("decrypting iam config: %w", err)
 	}
-	iamClientConfigMap, err := structToMap(iamClientConfig)
+	iamClientConfigMap, err := utils.StructToMap(iamClientConfig)
```

Comment:
rather than having `utils` package (which should be avoided in golang). why not adding a function receiver of the config, so converting the config to map would be something like:
`iamClientConfig.ToMap()`

Reply:
this is for avoiding redundancy of writing the same logic multiple times. In that case, we would implement the same logic for `x.ToMap()`, `y.ToMap()`, `z.ToMap()` and so on, and then when we want to refactor it, it would go back to the approach where we use the same shared function (`utils.StructToMap()`) all over again

Reply:
IMO it's still acceptable to have this shared function approach since the logics are exactly the same and not related with any domain or business context, it's purely a helper function. Based on what I see on existing packages for similar use cases, we can move this helper to `pkg/structs` instead, following golang's built-in `strings` helper fashion. WDYT?

there are 4 components (including 1 in a test package) used with `utils.StructToMap`

Reply:
yes we can do that later